sp1-prover/src/lib.rs

```diff
@@ -0,0 +1,133 @@
+use ecies::SecretKey;
+use serde::{Deserialize, Serialize};
+
+pub const PVK_HEX: &str = include_str!("../pvk");
+
+#[derive(Deserialize, Serialize)]
+pub struct AuctionData {
+    pub bidders: Vec<Bidder>,
+    pub id: Vec<u8>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct Bidder {
+    pub encrypted_amount: Vec<u8>,
+    pub address: Vec<u8>,
+}
+
+pub fn decrypt_bidder_data(pvk: &SecretKey, bidder: &Bidder) -> u128 {
+    let data = String::from_utf8(
+        ecies::decrypt(&pvk.serialize(), &bidder.encrypted_amount)
+            .expect("failed to decrypt"),
+    )
+        .unwrap();
+    data.parse().unwrap()
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs;
+    use crate::{AuctionData, Bidder, PVK_HEX};
+    // use rsa::pkcs8::{DecodePrivateKey, EncodePrivateKey, EncodePublicKey, LineEnding};
+    // use rsa::rand_core::OsRng;
+    // use rsa::{Pkcs1v15Encrypt, RsaPrivateKey, RsaPublicKey};
+    use sp1_sdk::{ProverClient, SP1Stdin};
+    // use std::env;
+    use std::fs::File;
+    use std::io::Read;
+    use std::process::Command;
+    use ecies::{PublicKey, SecretKey};
+    use rand::rngs::{OsRng};
+    // use ecies_ed25519::generate_keypair;
+    // use rand::{CryptoRng, RngCore};
+
+    #[test]
+    fn test_sp1_prover() {
+        // compile main
+        let output = Command::new("cargo")
```

Comment:
create a temp dir  https://docs.rs/tempdir/latest/tempdir/
and then build things on it